### ISSUE TITLE
uppercase -> title case broken

### ISSUE THREAD

**Issue by rpedela** [ianstormtaylor/to-case#5]
My use case is converting city and state names from 3rd-party data from uppercase to title case.

``` js
var toCase = require('to-case');

console.log(toCase.title('HENDERSON');
// EXPECT: Henderson
// RESULT: H E N D E R S O N

console.log(toCase.title('NEVADA');
// EXPECT: Nevada
// RESULT: N E v a D a
```

**Comment by tunnckoCore** [comment]
The bug comes from [uncamelize](https://github.com/ianstormtaylor/to-no-case/blob/master/index.js#L61) function from `to-no-case` package. Also applies for other case transforms, like sentence for example

``` js
var toCase = require('to-case');

console.log(toCase.sentence('FOO'); //=> F o o
console.log(toCase.sentence('HENDERSON'); //=> H e n d e r s o n
console.log(toCase.sentence('NEVADA'); //=> N e v a d a
```

Not works for single word strings in general.

**Comment by ianstormtaylor** [comment]
Happy to except a PR on `to-no-case` with a check to fix this!

**Comment by tunnckoCore** [comment]
Any ideas? Maybe need `isUppercased` and check `hasSpace` before passing it to `str.replace`?

**Comment by ianstormtaylor** [comment]
@tunnckoCore [This line](https://github.com/ianstormtaylor/to-no-case/blob/master/index.js#L26) probably needs to be behind an `if` check the same way the two above are, that checks to see if it's actually a camel-case string. And then if none of the `if`s pass it can just lowercase only I think.

Let me know if that makes sense!

**Comment by tunnckoCore** [comment]
Hm, both checks? Are really needed both? Because it looks like it works correctly now with only adding one `if`.

``` js
function toNoCase(string) {
  var _hasSpace = hasSpace.test(string)
  if (_hasSpace) return string.toLowerCase()
  if (hasSeparator.test(string)) return (unseparate(string) || string).toLowerCase()
  if (!_hasSpace) return string.toLowerCase()
  return uncamelize(string).toLowerCase()
}
```

and example

``` js
console.log(sentence('AN FOO AND A BAR')) //=> An foo and a bar
console.log(sentence('FOO')) //=> Foo
console.log(title('AN FOO AND A BAR')) //=> An Foo and a Bar
console.log(title('FOO')) //=> Foo
```

But I can't be sure in 100% that it's enough?

**Comment by ianstormtaylor** [comment]
That won't work for real camel-case strings though

**Comment by tunnckoCore** [comment]
Haha, yea.

Hm, I think I tricked it, all of the `to-case` tests are passing and also all `to-sentence-case`, all `to-no-case`, all `to-title-case`.

``` js
function toNoCase(str) {
  if (hasSpace.test(str)) {
    return str.toLowerCase()
  }
  if (hasSeparator.test(str)) {
    return (unseparate(str)|| str).toLowerCase()
  }
  if (!hasSpace.test(str) || isCamelCase(str)) {
    return uncamelize(str).toLowerCase()
  }
  return str.toLowerCase()
}
```

where `isCamelCase` is

``` js
function isCamelCase (str) {
  if (typeof str !== 'string') {
    throw new TypeError('is-camel-case: expect `str` be string')
  }
  if (str[0] !== str[0].toLowerCase()) return false
  if (/[^a-zA-Z]/.test(str)) return false
  if (str.toUpperCase() === str) return false
  if (str.toLowerCase() === str) return false

  var i = str.length
  var ret = false
  while (i--) {
    if (str[i].toUpperCase() === str[i]) {
      ret = true
      break
    }
  }
  return ret
}
```

where it pass

``` js
assert.strictEqual(isCamelCase('PascalCase'), false)
assert.strictEqual(isCamelCase('CAMELCASE'), false)
assert.strictEqual(isCamelCase('CAMELcase'), false)
assert.strictEqual(isCamelCase('camelCase'), true)  // notice
assert.strictEqual(isCamelCase('camelcasE'), true)  // notice
assert.strictEqual(isCamelCase('camel case'), false)
assert.strictEqual(isCamelCase('cAmeLCase'), true) // notice
assert.strictEqual(isCamelCase('camel-case'), false)
assert.strictEqual(isCamelCase('camel_case'), false)
```

Am I missing some "test" for it? I can publish it separately.

**Comment by tunnckoCore** [comment]
If that's okey we can continue to PRs.

**Comment by ianstormtaylor** [comment]
@tunnckoCore I was thinking something simpler, with a check to see if the string has a camel-case-like hump before `uncamelize`ing it. Like so: https://github.com/ianstormtaylor/to-no-case/blob/master/index.js#L14

Published `to-no-case@1.0.1` so this bug should be fixed!

**Comment by tunnckoCore** [comment]
Simpler? Regex is simpler and costs are lower than check most logical cases before do regex or looping? God sake. Meaning.. Above example of is camel case is more batter than anything else. Even `if (/[^a-zA-Z]/.test(str)) return false` isn't needed. Nevermind, you now best.

Okey, nevermind, cheers!
